doc/user/content/serve-results/s3.md

```diff
@@ -0,0 +1,233 @@
+   across all Materialize regions (see [AWS connection permissions](/sql/create-connection/#aws-permissions)).
+   For now, you'll set this ID to a dummy value; later, you'll update it with
+   the unique identifier for your Materialize region.
+
+1. Click **Next**.
+
+1. In **Add permissions**, select the IAM policy you created in [Create an IAM policy](#create-an-iam-policy),
+   and click **Next**.
+
+1. Enter a name for the role, and click **Create role**.
+
+1. Click **View role** to see the role summary page, and note down the
+   role **ARN**. You will need it in the next step to create an AWS connection in
+   Materialize.
+
+## Step 2. Create a connection
+
+1. In the [SQL Shell](https://console.materialize.com/), or your preferred SQL
+   client connected to Materialize, create an [AWS connection](/sql/create-connection/#aws),
+   replacing `<role>` with the name of the role you created in the previous
+   step:
+
+   ```sql
+   CREATE CONNECTION aws_connection
+      TO AWS (ASSUME ROLE ARN = 'arn:aws:iam::001234567890:role/<role>');
+   ```
+
+1. Retrieve the external ID for the connection:
+
+   ```sql
+   SELECT awsc.id, external_id
+    FROM mz_internal.mz_aws_connections awsc
+    JOIN mz_connections c ON awsc.id = c.id
+    WHERE c.name = 'aws_connection';
+   ```
+
+1. In your AWS account, find the IAM role you created in [Create an IAM role](#create-an-iam-role)
+   and, under **Trust relationships**, click **Edit trust policy**. Use the
+   `external_id` from the previous step to update the trust policy's
+   `ExternalId`, then click **Update policy**.
+
+   {{< warning >}}Failing to constrain the external ID in your role trust policy
+   will allow other Materialize customers to assume your role and use AWS
+   privileges you have granted the role!{{< /warning >}}
+
+1. Back in Materialize, validate the AWS connection you created using the
+   [`VALIDATE CONNECTION`](/sql/validate-connection) command.
+
+   ```sql
+   VALIDATE CONNECTION aws_connection;
+   ```
+
+   If no validation error is returned, you're ready to use this connection to
+   run a bulk export from Materialize to your target S3 bucket! 🔥
+
+## Step 4. Run a bulk export
+
+To export data to your target S3 bucket, use the [`COPY TO`](/sql/copy-to/#copy-to-s3)
+command, and the AWS connection you created in the previous step.
+
+{{< tabs >}}
+{{< tab "Parquet">}}
+
+```sql
+COPY some_object TO 's3://<bucket>/<path>'
+WITH (
+    AWS CONNECTION = aws_connection,
+    FORMAT = 'parquet'
+  );
+```
+
+For details on the Parquet writer settings Materialize uses, as well as data
+type support and conversion, check the [reference documentation](/sql/copy-to/#copy-to-s3-parquet).
+
+{{< /tab >}}
+
+{{< tab "CSV">}}
+
+```sql
+COPY some_object TO 's3://<bucket>/<path>'
+WITH (
+    AWS CONNECTION = aws_connection,
+    FORMAT = 'csv'
+  );
+```
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+You might notice that Materialize first writes a sentinel file to the target S3
```

Comment:
I think this needs more explanation, probably in the Snowflake guide, which currently doesn't mention the sentinel file. How would one use the delete event from the sentinel file to trigger an upload to snowflake?

Reply:
I don't agree that we need to explain further than this. We're already linking to the S3 docs for triggering things on event notifications, and their guides are going to be much better on explaining how to set up SNS/SQS to trigger processing on specific S3 event notifications. The processing system could be anything so I don't know whether we should be prescriptive on it.

Reply:
A lot of those guides show how to take action on object creation of the actual data file, not deletion of a sentinel file. The way we use the sentinel file is a bit particular to our implementation. We don’t need much, but some small example of what the delete event looks like and how to use it to get the actual data file. It seems obvious, but I think it’s worth mentioning this at least. Not a full on tutorial on setting up SNS.